tests/build/test_generate_config.py

```diff
@@ -499,3 +497,44 @@ def test_config_parsed_when_mbed_os_outside_project_root(program_in_mbed_os_subd
 
     assert "MBED_CONF_PLATFORM_STDIO_BAUD_RATE=9600" in config_text
     assert "MBED_LFS_READ_SIZE=64" in config_text
+
+
+def test_output_ext_unspecified(program):
+    generate_config("K64F", "GCC_ARM", program)
+    config_text = (program.files.cmake_build_dir / CMAKE_CONFIG_FILE).read_text()
+    hex_image = is_image_hex("K64F", program)
+
+    assert 'MBED_OUTPUT_EXT "hex"' not in config_text
+    assert hex_image is False
+
+
+def test_output_ext_bin(program):
+    target_data = TARGET_DATA.copy()
+    target_data["OUTPUT_EXT"] = "bin"
+
+    program.files.custom_targets_json.write_text(json.dumps({"IMAGINARYBOARD": target_data}))
+    target = "IMAGINARYBOARD"
```

Comment:
Should we swap these lines so you can dereference `target` in `program.files.custom_targets_json.write_text(json.dumps({"IMAGINARYBOARD": target_data}))`? This way you wouldn't have to write "IMAGINARYBOARD" twice.